### DISCUSSION TITLE
Hash compatibility matrix on game page (editable by dev+)

### DISCUSSION THREAD

**Discussion by rzumer** [RetroAchievements/RAWeb#789] [Ideas]
The idea is to clearly and efficiently indicate compatibility for an achievement set, by adding a 2D N1xN2 matrix where one dimension represents regions and the other versions. This would be displayed on the game page of every game with a core set published, with all cells blank by default, editable by developers and above.

Filling out the matrix should be a requirement for new sets.

Linked hacks, including fan translations, could be added as versions associated with the region over which they are patched, as compatibility is usually dependent on the base region.

In the below prototype, a set targets the JP region and is known to be incompatible with both US and EU regions. In the US and JP regions, two revisions exist, while in the other regions only a version equivalent to JP's 1.1 was published (open question: should they be labelled as 1.0 or 1.1 if they are the first version in their region, but equivalent to another region's version 1.1?). Region-version cells that do not correspond to an existing variant are left blank. A hypothetical retranslation patch applied over the JP version is available and also compatible, and a KR version exists, but was not tested.

![Prototype](https://user-images.githubusercontent.com/7488362/74046473-9e6eaa80-499c-11ea-9aad-56cd5afe3ebd.png)

Automatic matrix generation should also be relatively simple with this model, for systems that have databases with consistent naming standards (e.g. No-Intro).

Besides the three symbols shown in the prototype, an extra one for "partial compatibility" with an optional note displayed through tooltip or alt text could also be useful.

**Comment by luchaos** [comment]
Thank you for the input! I spent quite some time with the available no-intro datasets quite a while ago to find that the correlation of regional variants is nicely presented in those. An import, at least an initial one, is on the roadmap.

As for your question - `should they be labelled as 1.0 or 1.1 if they are the first version in their region, but equivalent to another region's version 1.1`
Now that's a hard one to answer - it all boils down to compatibility, right? Imo a mismatch in versions does not matter, compatibility comes first. I have yet to see how e.g. no-intro handles those, how well it  matches "memory compatibility" and how we might leverage them. By default - i'd opt for whatever version the database of our choice suggests. Are you aware of cases where relying on no-intro for that purpose might pose problems/ambiguities?

**Comment by Jamiras** [comment]
I believe this would be handled by #717, where "Current Rule" would be "Not Allowed"
